nilearn/plotting/surf_plotting.py

```diff
@@ -0,0 +1,376 @@
+"""
+Functions for surface visualization.
+Only matplotlib is required.
+"""
+
+from nilearn._utils.compat import _basestring
+from .img_plotting import _get_colorbar_and_data_ranges
+
+# Import libraries
+import numpy as np
+import nibabel
+from nibabel import gifti
+import matplotlib.pyplot as plt
+import matplotlib.tri as tri
+from mpl_toolkits.mplot3d import Axes3D
+
+
+# function to figure out datatype and load data
+def check_surf_data(surf_data):
+    # if the input is a filename, load it
+    if isinstance(surf_data, _basestring):
+        if (surf_data.endswith('nii') or surf_data.endswith('nii.gz') or
+                surf_data.endswith('mgz')):
+            data = np.squeeze(nibabel.load(surf_data).get_data())
+        elif (surf_data.endswith('curv') or surf_data.endswith('sulc') or
+                surf_data.endswith('thickness')):
+            data = nibabel.freesurfer.io.read_morph_data(surf_data)
+        elif surf_data.endswith('annot'):
+            data = nibabel.freesurfer.io.read_annot(surf_data)[0]
+        elif surf_data.endswith('label'):
+            data = nibabel.freesurfer.io.read_label(surf_data)
+        elif surf_data.endswith('gii'):
+            gii = gifti.read(surf_data)
+            data = np.zeros((len(gii.darrays[0].data), len(gii.darrays)))
+            for arr in range(len(gii.darrays)):
+                data[:, arr] = gii.darrays[arr].data
+            data = np.squeeze(data)
+        else:
+            raise ValueError('Format of data file not recognized.')
+    # if the input is a numpy array
+    elif isinstance(surf_data, np.ndarray):
+        data = np.squeeze(surf_data)
```

Comment:
Can we raise an error directly here rather than inside ?

```python
else:
      raise ValueError(Format of the given data file is not recognized. Expected file formats are .nii, .nii.gz, ........ You provided .format(surf_data))
```

From what I understood, only two types are accepted, either string or numpy array.